02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,334 @@
+
+## Prepare a final data frame with PCA results for plotting
+
+In the next chunk, we are going to extract the first two principal components from our `pca` object to prepare a data frame for plotting.
+
+```{r}
+# Make the first two principal components into a data frame for plotting with `ggplot2`
+pca_df <- data.frame(pca$x[, 1:2]) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame -- here we are going to join by `refinebio_accession_code` values
+  dplyr::inner_join(dplyr::select(metadata, refinebio_accession_code, histology, subgroup),
+    by = "refinebio_accession_code"
+  )
+```
+
+## Plot PCA Results
+
+Now let's plot the PC scores for the first two principal components since we know that they are responsible for the most explained proportion of variance in our dataset. 
+
+Let's also label the data points based on their genotype subgroup since medulloblastoma has been found to comprise of subgroups that each have molecularly distinct profiles [@Northcott2012].
+
+```{r}
+# Make a scatterplot using `ggplot2` functionality
+pca_plot <- ggplot(
+  pca_df,
+  aes(
+    x = PC1,
+    y = PC2,
+    color = subgroup # This will label points with different colors for each `subgroup`
+  )
+) +
+  geom_point() + # This tells R that we want a scatterplot
+  theme_classic() # This tells R to return a classic-looking plot with no gridlines
+
+# Print out plot here
+pca_plot
+```
+
+Looks like Group 4 and SHH groups somewhat cluster with each other but Group 3 seems to be less distinct as there are some samples clustering with Group 4 as well.
+
+We can add another label to our plot to get more information about our dataset.
+Let's also label the data points based on the histological subtype that each sample belongs to.
+
+```{r}
+# Make a scatterplot with ggplot2
+pca_plot <- ggplot(
+  pca_df,
+  aes(
+    x = PC1,
+    y = PC2,
+    color = subgroup, # This will label points with different colors for each `subgroup`
+    shape = histology # This will label points with different colors for each `histology` group
+  )
+) +
+  geom_point() +
+  theme_classic()
+
+# Print out plot here
+pca_plot
+```
+
+Adding the histological subtype label to our plot made our plot more informative, but was not helpful in distinguishing a better cluster behavior for Group 3.
+One may continue to add annotation labels to get a further understanding of the cluster behavior of each subgroup.
+
+## Save annotated PCA plot as a PNG
+
+Now that we have an annotated PCA plot, let's save it!
+
+You can easily switch this to save to a JPEG or TIFF by changing the file name within the `ggsave()` function to the respective file suffix.
+
+```{r}
+# Save plot using `ggsave()` function
+ggsave(file.path(
+  plots_dir,
+  "GSE37382_pca_scatterplot.png"
+), # Replace with name relevant your plotted data
```

Comment:
Can we move this comment to be in line with the line above?

Reply:
Oops, that change didn't save. Will make sure this is fixed in the next commit 👍